src/components/routes/Account/FirebaseAccountEmailHandler.tsx

```diff
@@ -29,42 +37,97 @@ const DefaultContainer = styled.div`
     section {
       margin-top: 18px;
     }
+
+    .input-container {
+      display: flex;
+      flex-direction: column;
+      margin: 16px 0px 28px 0px;
+      width: 100%;
+    }
   }
 `;
 
 export default class FirebaseAccountEmailHandler extends React.Component<RouterProps> {
   readonly state: State = {
-    isSubmitting: true,
+    isSubmitting: false,
     hasError: false,
-    errorMessage: ''
+    showPasswordResetForm: true,
+    showPasswordResetSuccess: false,
+    showVerifyEmailSuccess: false,
+    errorMessage: '',
+    successMessage: '',
+    password: '',
+    oobCode: ''
   };
 
   componentDidMount() {
     const queryParams: QueryParams = parseQueryString(this.props.location.search);
-     const { mode, oobCode } = queryParams;
-     if (!mode) {
-       alert('Error, no mode defined.');
-       return;
-     }
-
-     if (!oobCode) {
-       alert('Error, no oobCode defined.');
-       return;
-     }
-
-     if (mode === 'verifyEmail') {
-       verifyEmail(oobCode).then(() => {
-         this.setState({ isSubmitting: false });
-       }).catch(err => {
-         console.error(err);
-         this.setState({ isSubmitting: false, hasError: true });
-       });
-     }
+    const { mode, oobCode } = queryParams;
+
+    if (!mode) {
+      alert('Error, no mode defined.');
+      return;
+    }
+
+    if (!oobCode) {
+      alert('Error, no oobCode defined.');
+      return;
+    }
+    this.setState({oobCode});
+
+    // @see https://firebase.google.com/docs/auth/custom-email-handler
+    switch (mode) {
+     case 'resetPassword':
+        this.handleResetPassword(oobCode);
+        break;
+     case 'verifyEmail':
+        this.handleVerifyEmail(oobCode);
+       break;
+     default:
+       // Error: invalid mode.
+    }
+  }
+
+  handleFormChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = event.target;
+    return this.setState({[name]: value });
+  }
+
+  handleVerifyEmail(oobCode:string) {
+    auth.applyActionCode(oobCode).then(() => {
+      this.setState({ isSubmitting: false, showVerifyEmailSuccess: true, successMessage: 'Thanks for verifying your email.' });
```

Comment:
should we move these success/error messages to utils?

Reply:
i think this is fine since these two strings are only referenced here and no where else.